clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -365,6 +365,10 @@ private KafkaChannel buildAndAttachKafkaChannel(SocketChannel socketChannel, Str
             // Ideally, these resources are closed by the KafkaChannel but if the KafkaChannel is not created to an
             // error, this builder should close the resources it has created instead.
             Utils.closeQuietly(metadataRegistry, "metadataRegistry");
+            // If an Authentication Failure is encountered, throw that error directly.
+            if (e instanceof AuthenticationException) {
```

Comment:
should this just be for AuthN or any fatal in general? e.g. AuthorizationException should also be treated similarily.